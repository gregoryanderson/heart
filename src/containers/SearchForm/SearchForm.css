.SearchForm {
    text-align: center;
}

.search-text {
    padding: 25px;
}

.search {
    width: 200px;
}