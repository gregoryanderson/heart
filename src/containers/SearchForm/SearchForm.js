import React, { Component } from "react";
import "./SearchForm.css";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import Wall from "../Wall/Wall";
import { Link } from "react-router-dom";
import FacetContainer from "../FacetContainer/FacetContainer";

export class SearchForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentValue: null
    };
  }

  handleChangeOfInput = e => {
    this.setState({ currentValue: e.target.value });
  };

  render() {
    return (
      <section className="SearchForm">
        <p className="search-text">Please choose from one of these options:</p>
        <FacetContainer
          handleSearch={this.props.handleSearch}
          name={this.props.name}
          facets={this.props.facets}
        />
        <p className="search-text">Or ask for something more specific..</p>
        <form>
          <input
            type="text"
            placeholder={this.props.placeholder}
            name={this.props.name}
            className="search"
            value={this.state.name}
            onChange={e => this.handleChangeOfInput(e)}
          ></input>
          <Link
            to={`/${this.props.name}/${this.state.currentValue}`}
            onClick={e =>
              this.props.handleSearch(this.props.name, this.state.currentValue)
            }
          >
            <p className="search-text">..and find some paintings</p>
          </Link>
        </form>
        <Wall
          paintings={this.props.paintings}
          handleFavorite={this.props.handleFavorite}
          route="favorites"
        />
      </section>
    );
  }
}

export const mapStateToProps = state => ({
  paintings: state.paintings,
  favorites: state.favorites
});

export default connect(mapStateToProps)(SearchForm);

SearchForm.propTypes = {
  addFavoriteInRedux: PropTypes.func,
  deleteFavoriteInRedux: PropTypes.func,
  establishFacetsInRedux: PropTypes.func,
  establishPaintingsInRedux: PropTypes.func,
  facets: PropTypes.object,
  favorites: PropTypes.array,
  paintings: PropTypes.array,
  name: PropTypes.string
};
