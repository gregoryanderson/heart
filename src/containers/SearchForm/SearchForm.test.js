import React from "react";
import { shallow } from "enzyme";
import { SearchForm, mapStateToProps } from "./SearchForm";

describe("SearchForm", () => {
  let wrapper,
    mockFacets,
    mockFavorites,
    mockHandleFavorite,
    mockHandleSearch,
    mockName,
    mockPaintings,
    mockPlaceholder;

  beforeEach(() => {
    mockFacets = {
      facets: [
        {
          0: {},
          1: {}
        }
      ]
    };

    mockPaintings = [
      {
        title: "Clock and gunpowder horn",
        id: "en-NG-NM-7687",
        artist: "anonymous",
        width: 1672,
        height: 2500,
        url:
          "https://lh3.ggpht.com/lAJ1wnr_hEOncOfh9eKzvaS8w-fhLLq5yGlzHBctnjgyOzsbuP4cGIqP4q0A-YvnyXBhJi96il6NIZNhRVW-BVg2lW0=s0",
        isFav: false
      }
    ];

    mockFavorites = [
      {
        title: "Cupboard",
        id: "en-BK-1975-81",
        artist: "Herman Doomer",
        width: 2026,
        height: 2500,
        url:
          "https://lh3.googleusercontent.com/tGI4dOAfJLBbewwspzXpUnSZxEKFACv9Y3FHqAxQUtN2p4AXt2MS9oFv6eJyIBtr7gvzmv58vSitMFVeHY0TGsfOfDN2=s0",
        isFav: false
      }
    ];

    mockName = "artist";
    mockPlaceholder = "Please select an artist";

    mockHandleFavorite = jest.fn();
    mockHandleSearch = jest.fn();

    wrapper = shallow(
      <SearchForm
        handleSearch={mockHandleSearch}
        handleFavorite={mockHandleFavorite}
        paintings={mockPaintings}
        favorites={mockFavorites}
        facets={mockFacets}
      />
    );
  });
  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should handle the change of input", () => {
    mockHandleChange = handleChangeOfInput;

    mockHandleChange = jest.fn();
    wrapper.find("input").simulate("change");
    expect(mockHandleChange).toHaveBeenCalled();
  });
});

describe("mapStateToProps", () => {
  it("should properly map to props", () => {
    const mockFavorites = [
      {
        title: "Cupboard",
        id: "en-BK-1975-81",
        artist: "Herman Doomer",
        width: 2026,
        height: 2500,
        url:
          "https://lh3.googleusercontent.com/tGI4dOAfJLBbewwspzXpUnSZxEKFACv9Y3FHqAxQUtN2p4AXt2MS9oFv6eJyIBtr7gvzmv58vSitMFVeHY0TGsfOfDN2=s0",
        isFav: false
      }
    ];
    const mockPaintings = [
      {
        title: "Clock and gunpowder horn",
        id: "en-NG-NM-7687",
        artist: "anonymous",
        width: 1672,
        height: 2500,
        url:
          "https://lh3.ggpht.com/lAJ1wnr_hEOncOfh9eKzvaS8w-fhLLq5yGlzHBctnjgyOzsbuP4cGIqP4q0A-YvnyXBhJi96il6NIZNhRVW-BVg2lW0=s0",
        isFav: false
      }
    ];

    const mockState = {
      favorites: [
        {
          title: "Cupboard",
          id: "en-BK-1975-81",
          artist: "Herman Doomer",
          width: 2026,
          height: 2500,
          url:
            "https://lh3.googleusercontent.com/tGI4dOAfJLBbewwspzXpUnSZxEKFACv9Y3FHqAxQUtN2p4AXt2MS9oFv6eJyIBtr7gvzmv58vSitMFVeHY0TGsfOfDN2=s0",
          isFav: false
        }
      ],
      paintings: [
        {
          title: "Clock and gunpowder horn",
          id: "en-NG-NM-7687",
          artist: "anonymous",
          width: 1672,
          height: 2500,
          url:
            "https://lh3.ggpht.com/lAJ1wnr_hEOncOfh9eKzvaS8w-fhLLq5yGlzHBctnjgyOzsbuP4cGIqP4q0A-YvnyXBhJi96il6NIZNhRVW-BVg2lW0=s0",
          isFav: false
        }
      ]
    };

    expect(mapStateToProps(mockState).paintings).toEqual(mockPaintings);
    expect(mapStateToProps(mockState).favorites).toEqual(mockFavorites);
  });
});

describe("mapDispatchToProps", () => {
    it("should properly dispatch to props", () => {
      const mockFavorites = [
        {
          title: "Cupboard",
          id: "en-BK-1975-81",
          artist: "Herman Doomer",
          width: 2026,
          height: 2500,
          url:
            "https://lh3.googleusercontent.com/tGI4dOAfJLBbewwspzXpUnSZxEKFACv9Y3FHqAxQUtN2p4AXt2MS9oFv6eJyIBtr7gvzmv58vSitMFVeHY0TGsfOfDN2=s0",
          isFav: false
        }
      ];
      const mockPaintings = [
        {
          title: "Clock and gunpowder horn",
          id: "en-NG-NM-7687",
          artist: "anonymous",
          width: 1672,
          height: 2500,
          url:
            "https://lh3.ggpht.com/lAJ1wnr_hEOncOfh9eKzvaS8w-fhLLq5yGlzHBctnjgyOzsbuP4cGIqP4q0A-YvnyXBhJi96il6NIZNhRVW-BVg2lW0=s0",
          isFav: false
        }
      ];
  
      const mockState = {
        favorites: [
          {
            title: "Cupboard",
            id: "en-BK-1975-81",
            artist: "Herman Doomer",
            width: 2026,
            height: 2500,
            url:
              "https://lh3.googleusercontent.com/tGI4dOAfJLBbewwspzXpUnSZxEKFACv9Y3FHqAxQUtN2p4AXt2MS9oFv6eJyIBtr7gvzmv58vSitMFVeHY0TGsfOfDN2=s0",
            isFav: false
          }
        ],
        paintings: [
          {
            title: "Clock and gunpowder horn",
            id: "en-NG-NM-7687",
            artist: "anonymous",
            width: 1672,
            height: 2500,
            url:
              "https://lh3.ggpht.com/lAJ1wnr_hEOncOfh9eKzvaS8w-fhLLq5yGlzHBctnjgyOzsbuP4cGIqP4q0A-YvnyXBhJi96il6NIZNhRVW-BVg2lW0=s0",
            isFav: false
          }
        ]
      };
  
      expect(mapStateToProps(mockState).paintings).toEqual(mockPaintings);
      expect(mapStateToProps(mockState).favorites).toEqual(mockFavorites);
    });
  });
  