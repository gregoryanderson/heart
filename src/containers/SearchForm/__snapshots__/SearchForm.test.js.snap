// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot 1`] = `
<section
  className="SearchForm"
>
  <p
    className="search-text"
  >
    Please choose from one of these options:
  </p>
  <FacetContainer
    facets={
      Object {
        "facets": Array [
          Object {
            "0": Object {},
            "1": Object {},
          },
        ],
      }
    }
    handleSearch={[MockFunction]}
  />
  <p
    className="search-text"
  >
    Or ask for something more specific..
  </p>
  <form>
    <input
      className="search"
      onChange={[Function]}
      type="text"
    />
    <Link
      onClick={[Function]}
      to="/undefined/null"
    >
      <p
        className="search-text"
      >
        ..and find some paintings
      </p>
    </Link>
  </form>
  <Wall
    handleFavorite={[MockFunction]}
    paintings={
      Array [
        Object {
          "artist": "anonymous",
          "height": 2500,
          "id": "en-NG-NM-7687",
          "isFav": false,
          "title": "Clock and gunpowder horn",
          "url": "https://lh3.ggpht.com/lAJ1wnr_hEOncOfh9eKzvaS8w-fhLLq5yGlzHBctnjgyOzsbuP4cGIqP4q0A-YvnyXBhJi96il6NIZNhRVW-BVg2lW0=s0",
          "width": 1672,
        },
      ]
    }
    route="favorites"
  />
</section>
`;
